src/balloonpanel/balloonpanelview.js

```diff
@@ -305,43 +301,55 @@ function getAbsoluteRect( elementOrRangeOrRect ) {
 	return absoluteRect;
 }
 
-/**
- * Returns the client rect of the element limited by the visible (to the user)
- * viewport of the browser window.
- *
- *		[Browser viewport]
- *		+---------------------------------------+
- *		|                        [Element]      |
- *		|                        +----------------------+
- *		|                        |##############|       |
- *		|                        |##############|       |
- *		|                        |#######^######|       |
- *		|                        +-------|--------------+
- *		|                                |      |
- *		+--------------------------------|------+
- *		                                 |
- *		                                  \- [Element rect visible in the viewport]
- *
- * @private
- * @param {HTMLElement|Object} element Object which visible area rect is to be determined.
- * @returns {AbsoluteDomRect} An absolute rect of the area visible in the viewport.
- */
-function getAbsoluteRectVisisbleInTheViewport( element ) {
-	const limiterRect = new AbsoluteDomRect( element, 'limiterElement' );
-
-	const windowScrollX = window.scrollX;
-	const windowScrollY = window.scrollY;
-	const bodyWidth = document.body.clientWidth;
-	const bodyHeight = document.body.clientHeight;
+// Returns the client rect of the element limited by the visible (to the user)
+// viewport of the browser window.
+//
+//		[Browser viewport]
+//		+---------------------------------------+
+//		|                        [Element]      |
+//		|                        +----------------------+
+//		|                        |##############|       |
+//		|                        |##############|       |
+//		|                        |#######^######|       |
+//		|                        +-------|--------------+
+//		|                                |      |
+//		+--------------------------------|------+
+//		                                 |
+//		                                  \- [Element rect visible in the viewport]
+//
+// @private
+// @param {HTMLElement|Object} element Object which visible area rect is to be determined.
+// @returns {AbsoluteDomRect} An absolute rect of the area visible in the viewport.
+function getAbsoluteRectVisibleInTheViewport( element ) {
+	const limiterRect = getAbsoluteRect( element );
+	const viewportRect = getViewportRect();
 
 	return new AbsoluteDomRect( {
-		top: Math.max( limiterRect.top, windowScrollY ),
-		left: Math.max( limiterRect.left, windowScrollX ),
-		right: Math.min( limiterRect.right, bodyWidth + windowScrollX ),
-		bottom: Math.min( limiterRect.bottom, bodyHeight + windowScrollY )
+		top: Math.max( limiterRect.top, viewportRect.top ),
+		left: Math.max( limiterRect.left, viewportRect.left ),
+		right: Math.min( limiterRect.right, viewportRect.right ),
+		bottom: Math.min( limiterRect.bottom, viewportRect.bottom )
 	} );
 }
 
+// Get browser viewport rect.
+//
+// @private
+// @returns {Object} Viewport rect.
+function getViewportRect() {
```

Comment:
What sort of a rect is it? An absolute rect or `getBoundingClientRect`? The name of the function should tell this because it really matters.